src/opcodes/tests/test_opcodes.cairo

```diff
@@ -362,6 +362,107 @@ fn test_op_depth_empty_stack() {
+    let mut engine = EngineTraitImpl::new(bytecode);
+    let _ = engine.step();
+    let _ = engine.step();
+    let _ = engine.step();
+    let res = engine.step();
+    assert!(res, "Execution of step failed");
+
+    let dstack = engine.get_dstack();
+    assert_eq!(dstack.len(), 3, "Stack length is not 1");
+
+    let expected_stack = array!["\x02", "\x03", "\x01"];
+    assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
+}
+
+#[test]
+fn test_op_swap_mid() {
+    let program = "OP_1 OP_2 OP_3 OP_SWAP OP_4 OP_5";
+    let mut compiler = CompilerTraitImpl::new();
+    let bytecode = compiler.compile(program);
+    let mut engine = EngineTraitImpl::new(bytecode);
+    let _ = engine.step();
+    let _ = engine.step();
+    let _ = engine.step();
+    let _ = engine.step();
+    let _ = engine.step();
+    let res = engine.step();
+    assert!(res, "Execution of step failed");
+
+    let dstack = engine.get_dstack();
+    assert_eq!(dstack.len(), 5, "Stack length is not 1");
+
+    let expected_stack = array!["\x05", "\x04", "\x02", "\x03", "\x01"];
+    assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
+}
+
+
+#[test]
+fn test_op_2swap() {
+    let program = "OP_1 OP_2 OP_3 OP_4 OP_2SWAP";
+    let mut compiler = CompilerTraitImpl::new();
+    let bytecode = compiler.compile(program);
+    let mut engine = EngineTraitImpl::new(bytecode);
+    let _ = engine.step(); //push 1
+    let _ = engine.step(); //push 2
+    let _ = engine.step(); //push 3
+    let _ = engine.step(); //push 4
+    let dstack = engine.get_dstack();
+    let expected_stack = array!["\x04", "\x03", "\x02", "\x01"];
+    assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
+
+    let res = engine.step(); //execute op_2swap
+    assert!(res, "Execution of step failed");
+
+    let dstack = engine.get_dstack();
+    assert_eq!(dstack.len(), 4, "Stack length is not 1");
+
+    let expected_stack = array!["\x02", "\x01", "\x04", "\x03"];
+    assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
+}
+
+#[test]
+fn test_op_2swap_mid() {
+    let program = "OP_1 OP_2 OP_3 OP_4 OP_2SWAP OP_5 OP_6";
+    let mut compiler = CompilerTraitImpl::new();
+    let bytecode = compiler.compile(program);
+    let mut engine = EngineTraitImpl::new(bytecode);
+    let _ = engine.step(); //push 1
+    let _ = engine.step(); //push 2
+    let _ = engine.step(); //push 3
+    let _ = engine.step(); //push 4
+    let _ = engine.step(); //execute op_2swap
+    let _ = engine.step(); //push 5
+    let res = engine.step(); //push 6
+    let dstack = engine.get_dstack();
+
+    assert!(res, "Execution of step failed");
+    assert_eq!(dstack.len(), 6, "Stack length is not 1");
+
+    let expected_stack = array!["\x06", "\x05", "\x02", "\x01", "\x04", "\x03"];
+    assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
+}
+
+#[test]
+fn test_op_2swap_underflow() {
```

Comment:
For this test, I think you will want to use the `#[should_panic]` declaration and do one more ` let _ = engine.step();` line to cause the `OP_2SWAP` overflow : https://book.cairo-lang.org/ch10-01-how-to-write-tests.html#checking-for-panics-with-should_panic

Reply:
Understood and implemented the same